VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -2,32 +2,52 @@
 
 import android.content.res.Resources;
 import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
 import android.text.Html;
-import android.text.Spanned;
-import android.text.TextUtils;
 import android.text.format.DateUtils;
 
 import org.edx.mobile.R;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.util.HashMap;
 
 public abstract class DiscussionTextUtils {
     private DiscussionTextUtils() {
     }
 
-    public static String getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources) {
-        String text = authorData.getAuthor() + " " + DateUtils.getRelativeTimeSpanString(
+    public static CharSequence getAuthorAttributionText(@NonNull DiscussionThread discussionThread, @NonNull Resources resources) {
```

Comment:
Also, it would probably be more convenient if these methods accepted a `Context` instance instead of `Resources`, as the latter can usually only be accessed from the former and is usually already available as a variable anyway.